### ISSUE TITLE
Problems on Heroku

### ISSUE THREAD

**Issue by brandonweiss** [juliocesar/rack-pagespeed#22]
So, I've got a small sinatra app on Heroku, and recently I added in rack-pagespeed. This is what I put in my config.ru.

```
use Rack::PageSpeed, :public => "./public" do
  store :disk => Dir.tmpdir # require 'tmpdir'
  combine_css
end
```

Everything was cool at first, but then like a week or two later I go to my site and find it unstyled. The link that rack-pagespeed was generating for the combined CSS was bad. But I thought this was my fault at first. I know Heroku has problems with relative paths some time, so I assumed the problem was the `./public` and that I just hadn't checked it after deploying initially.

So I changed it to this:

```
use Rack::PageSpeed, :public => "#{Minimalog.settings.root}/public" do
  store :disk => Dir.tmpdir # require 'tmpdir'
  combine_css
end
```

And in the sinatra app I set the root like this:

```
set :root, File.dirname(__FILE__)
```

And this worked for a little while, until today, when I go to the site and find it unstyled again. Any idea what might be going wrong? Or what I might be doing wrong? Thanks!

**Comment by juliocesar** [comment]
Heroku wipes the app's tmpdir every once in a while. That's what's happening. You shouldn't rely on disk storage when using it.

Try `:memcached` when on Heroku. You need to enable the addon first.

**Comment by mattash** [comment]
I'm having problems on Heroku as well, while using memcached. The error is as follows:

```
2011-04-08T13:34:59+00:00 app[web.1]: /app/.bundle/gems/ruby/1.8/gems/activesupport-3.0.5/lib/active_support/dependencies.rb:235:in `load': no such file to load -- /app/.bundle/gems/ruby/1.8/gems/rack-pagespeed-1.0.6/lib/rack/pagespeed/store/memcached (LoadError)
```

Could this be a 1.8.7 vs. 1.9.x issue? I'm on the bamboo-ree-1.8.7 stack.

**Comment by juliocesar** [comment]
No, couldn't. It's a problem I've introduced in the last released with the dynamic store loader. Fixed now in 1.0.7. Thanks!
